node/lib/util/commit.js

```diff
@@ -257,6 +307,17 @@ exports.commit = co.wrap(function *(metaRepo, all, metaStatus, message) {
     const subCommits = {};
     let subsChanged = false;
     const commitSubmodule = co.wrap(function *(name) {
```

Comment:
Only subCommits and subsChanged are written to -- and the subsChanged section could be broken out into a separate function.  So I think it would be possible (and clearer) to break this out.